### ISSUE TITLE
Sentence embeddings for author/news source attribution

### ISSUE THREAD

**Issue by ericlief** [UKPLab/sentence-transformers#1182]
We have been experimenting with different setups for the task of news source verification. Our first approach trained on sentence pairs from same and different source domains with cosine loss. For verification, mean cosine distance was used to measure distance from a random probe and a all embeddings for each news source profile profile. We achieved around 25% EER. Part of the problem was that texts from a source came from a variety of topics and were written by numerous and often unkown authors. I recently obtained good results using a softmax transformer to classify the relevant sources. However, using a handful of common pooling strategies does not yield good performance in verification (higher than expected cosine scores with normalization). I have read other posts here and elsewhere  and realize there are other losses, but is it possible to use traditional softmax to learn complicated stylistic features rather than semantic ones, and stil measure distance in some linear space? Do you know of other lower level embeddings that might be extracted (The last two layers haven't worked so well..)? 

I wanted to finetune my classification model with sbert but coudnt't figure out how to use my model under the hood without having to mess a lot with your code (I think the different configs were the problem).

**Comment by nreimers** [comment]
I sadly not sure what want to achieve.

In principle, what you need are positive pairs (text1, text2), which you want to have close in the vector space. You can then train with MultipleNegativesRankingLoss. 

The positive pair can be of anything, e.g. semantically similar content, or content with the same stylistic features etc., it does not really matter.

**Comment by ericlief** [comment]
Thanks for your qucik answer @nreimers! I will look into this type of objective--I assume you mean to cluster in an unsupervised manner the data from the different known news sources? I would choose positive duplicate text pairs of the articles and negative examples would automatically be created from the rest of the training data? Or should I also provide mismatching texts (like we did for the cosine objective, e.g. (new york times, bbc)?
As for the issue, I have with the config from training a HF model for a classification task (producing a config different from that of sbert):
{
  "_name_or_path": "sentence-transformers/all-distilroberta-v1",
  "architectures": [
    "RobertaForSequenceClassification"
  ],
  "attention_probs_dropout_prob": 0.1,
  "bos_token_id": 0,
  "classifier_dropout": null,
  "eos_token_id": 2,
  "gradient_checkpointing": false,
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 768,
  "id2label": {
    "0": "LABEL_0",
    "1": "LABEL_1",
    "2": "LABEL_2",
    "3": "LABEL_3",
    "4": "LABEL_4",
    "5": "LABEL_5",
    "6": "LABEL_6",
    "7": "LABEL_7",
    "8": "LABEL_8",
    "9": "LABEL_9"
  },
  "initializer_range": 0.02,
  "intermediate_size": 3072,
  "label2id": {
    "LABEL_0": 0,
    "LABEL_1": 1,
    "LABEL_2": 2,
    "LABEL_3": 3,
    "LABEL_4": 4,
    "LABEL_5": 5,
    "LABEL_6": 6,
    "LABEL_7": 7,
    "LABEL_8": 8,
    "LABEL_9": 9
  },
  "layer_norm_eps": 1e-05,
  "max_position_embeddings": 514,
  "model_type": "roberta",
  "num_attention_heads": 12,
  "num_hidden_layers": 6,
  "pad_token_id": 1,
  "position_embedding_type": "absolute",
  "problem_type": "single_label_classification",
  "torch_dtype": "float32",
  "transformers_version": "4.10.0",
  "type_vocab_size": 1,
  "use_cache": true,
  "vocab_size": 50265
}
I do not know if I can vice versa train a vanilla transformer model on the hub (e.g. distilroberta) and load as an sbert model to train. This was my plan with the classification model--to use hidden states and possibly finetune with some loss other than softmax on the data
